url/src/parser.rs

```diff
@@ -1354,9 +1353,56 @@ impl<'a> Parser<'a> {
         host_end: u32,
         host: HostInternal,
         port: Option<u16>,
-        path_start: u32,
+        mut path_start: u32,
         remaining: Input<'_>,
     ) -> ParseResult<Url> {
+        // Special case for anarchist URL's with a leading empty fragment
```

Comment:
fragment has a special meaning in the context of a URL - it's the bit after the # sign. Let's use another word here instead.

Reply:
Done